availability-prober/availability_prober.go

```diff
@@ -146,6 +151,29 @@ func check(log logr.Logger, target *url.URL, requestTimeout time.Duration, sleep
 			}
 		}
 
+		if waitForLabeledPodsGone != "" {
+			namespace := strings.Split(waitForLabeledPodsGone, "/")[0]
+			labelSelectors := strings.TrimPrefix(waitForLabeledPodsGone, fmt.Sprintf("%s/", namespace))
+			pods := &corev1.PodList{}
+			labelSet, err := labels.ConvertSelectorToLabelsMap(labelSelectors)
+			if err != nil {
+				log.Error(err, fmt.Sprintf("invalid label selectors %s", labelSelectors))
+				continue
+			}
+			err = kubeClient.List(context.Background(), pods, &crclient.ListOptions{
+				Namespace:     namespace,
+				LabelSelector: labels.SelectorFromValidatedSet(labelSet),
+			})
+			if err != nil && !apierrors.IsNotFound(err) {
+				log.Error(err, fmt.Sprintf("failed to get pods with label %s in namespace %s, retrying...", labelSelectors, namespace))
+				continue
+			}
+			if pods != nil && len(pods.Items) != 0 {
+				log.Info(fmt.Sprintf("pods with label %s still exist in namespace %s, retrying...", labelSelectors, namespace))
```

Comment:
We need to be able to handle if the pods are stuck in terminating in the cluster and not block rollout

Reply:
So it should be pods not there first and then if the deletionTimestamp is more than 10 minutes old it’s safe to continue (if it’s on a dead node it will stay in terminating)

Reply:
Ah that's a very good point and idea. Will get updates pushed.

Reply:
Updated.